### PR TITLE
Chore: Use jsoniter in cloud-monitoring from grafana-plugin-sdk-go

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

We have jsoniter package in `grafana-plugin-sdk-go`. No need to have it in `grafana/grafana`

**Why do we need this feature?**

Just cleaning up

I created separate PRs for each data source. If something goes wrong we can revert individually. 
- [Prometheus](https://github.com/grafana/grafana/pull/82359)
- [InfluxDB](https://github.com/grafana/grafana/pull/82360)
- [Cloud Monitoring](https://github.com/grafana/grafana/pull/82358)

### CODE REVIEW

File: pkg/tsdb/cloud-monitoring/converter/converter.go [hunk 1]
```diff
@@ -566,12 +564,16 @@ func readScalar(iter *jsonitere.Iterator) backend.DataResponse {
 		Custom: resultTypeToCustomMeta("scalar"),
 	}
 
+	if dataPlane {
```
Comment (aangelisc) [inline pkg/tsdb/cloud-monitoring/converter/converter.go:567]: Why is this change being made?

**Review by aangelisc — APPROVED** [review]
LGTM just one question in the comments 😊

File: pkg/tsdb/cloud-monitoring/converter/converter.go (hunk 1 above)
Comment (itsmylife) [reply pkg/tsdb/cloud-monitoring/converter/converter.go:567]: https://github.com/grafana/grafana/pull/81491

Comment (aangelisc) [reply pkg/tsdb/cloud-monitoring/converter/converter.go:567]: Thank you for clarifying! All good to merge 😊

**Pull request merged**